schema/deploy/functions/handle_default_form_change_commit.sql

```diff
@@ -38,12 +39,31 @@ begin
       select column_name::text
         from information_schema.columns
         where table_schema = fc.form_data_schema_name
-        AND table_name   = fc.form_data_table_name
+        and table_name   = fc.form_data_table_name
     ) then
       json_data = json_data - json_key;
     end if;
   end loop;
 
+  /**
+  we need to add null values for any columns that are not in the form_data object
+  this is necessary because we may remove an optional field from the form but we don't update the data in the table because it is not in the form_data object
+  We don't want this to happen when operation is create since we need to use the default values provided by the table
+  **/
+  if fc.operation = 'update' then
+    for table_column_name in (
+        select column_name::text
+        from information_schema.columns
+        where table_schema = fc.form_data_schema_name
+          and table_name = fc.form_data_table_name
+          and (select cif_private.snake_to_camel_case(column_name)) not in (select key from jsonb_each(json_data))
+          and column_name::text not in ('id', 'created_at', 'updated_at', 'archived_at')
```

Comment:
I think we also don't want to wipe `created_by`, `updated_by`, `archived_by`. I'm trying to think if there's a way to continuously check for columns like this. If we add a column to a table in the future but don't remember to update this function, we'll unintentionally delete data